vault/logical_cubbyhole.go

```diff
@@ -140,17 +141,22 @@ func (b *CubbyholeBackend) handleList(
 	if req.ClientToken == "" {
 		return nil, fmt.Errorf("[ERR] cubbyhole list: Client token empty")
 	}
+
 	// List the keys at the prefix given by the request
 	keys, err := req.Storage.List(req.ClientToken + "/" + req.Path)
 	if err != nil {
 		return nil, err
 	}
 
-	strippedKeys := []string{}
-	for _, key := range keys {
-		strippedKeys = append(strippedKeys, strings.TrimPrefix(key, req.ClientToken+"/"))
+	// Strip the token; also, add the path for the same reason as in
+	// passthrough
+	strippedKeys := make([]string, len(keys))
+	for i, key := range keys {
+		strippedKeys[i] = req.MountPoint + req.Path + strings.TrimPrefix(key, req.ClientToken+"/")
 	}
 
+	sort.Strings(strippedKeys)
```

Comment:
I don't know if we want it or not. Maybe the best path is to sort the output in the CLI, but otherwise leave it up to clients, rather than take the speed hit ourselves. I'll do that for now.